source/algorithms.tex

```diff
@@ -3540,17 +3540,18 @@
   for the overloads with both parameters \tcode{pred} and \tcode{proj1}.
 \end{itemize}
 
+\pnum
+Let $N$ be $\min(\tcode{last1 - first1}, \ \tcode{last2 - first2})$.
+
 \pnum
 \returns
 \tcode{\{ first1 + n, first2 + n \}},
 where \tcode{n} is the smallest integer such that $E$ holds,
-or $\min(\tcode{last1 - first1}, \ \tcode{last2 - first2})$
-if no such integer exists.
+or $N$ if no such integer less than $N$ exists.
```

Comment:
This wording is awkward. It's much more common in mathematics to say precisely what you want the first time round, rather than adding constraints later:

>    where `n` is the smallest integer less than $N$ such that $E$ holds, or ... if no such integer exists.

Reply:
It's also weird that $E$ uses `n` but doesn't formally depend on it; I would much prefer to call it $E(n)$. Then, if you move the definition of $N$ up, you can even say that "$E(N)$ is true" (or rather, "$E(n) is ... true when n is N").

Reply:
Fixed the first issue. Left the second issue alone; the pattern that the index variable is not a parameter of E is pervasive (see e.g. [alg.any_of] and friends).